<launch>

  <!-- joint 1 limits params -->
  <param name="joint1/limit/lower" type="double" value="-3.14" />
  <param name="joint1/limit/upper" type="double" value="3.14" />
  <!-- joint 2 limits params -->
  <param name="joint2/limit/lower" type="double" value="-0.90" />
  <param name="joint2/limit/upper" type="double" value="1.3963" />
  <!-- joint 3 limits params -->
  <param name="joint3/limit/lower" type="double" value="-2.5" />
  <param name="joint3/limit/upper" type="double" value="0" />
  <!-- joint 4 limits params -->
  <param name="joint4/limit/lower" type="double" value="-3.14" />
  <param name="joint4/limit/upper" type="double" value="3.14" />
  <!-- joint 5 limits params -->
  <param name="joint5/limit/lower" type="double" value="0" />
  <param name="joint5/limit/upper" type="double" value="3.14" />
  <!-- joint 6 limits params -->
  <param name="joint6/limit/lower" type="double" value="-3.14" />
  <param name="joint6/limit/upper" type="double" value="3.14" />

  <arg name="model" />
  <arg name="gui" default="True" />
  <param name="robot_description" textfile="$(find manipulator)/urdf/manipulator_effort.urdf" />

  <node name="robot_state_publisher"
    pkg="robot_state_publisher"
    type="robot_state_publisher"
    respawn="false"
    output="screen">
  </node>

  <node pkg="tf" type="static_transform_publisher" name="base_to_world"
        args="0 0 0 0 0 0  /base_link /world 100" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find manipulator)/urdf/urdf.rviz -f base_link" />

  <node name="kdl_dkin" pkg="manipulator" type="kdl_dkin" output="screen"></node>

  <param name="use_gui" value="$(arg gui)" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" ></node>

</launch>
